invenio_search/api.py

```diff
@@ -267,18 +267,19 @@ def get_records(self, ids):
         """
         return self.query(Ids(values=[str(id_) for id_ in ids]))
 
-    def with_preference_param(self, preference):
+    def with_preference_param(self, preference=None):
         """Add the preference param to the ES request and return a new Search.
 
         The preference param avoids the bouncing effect with multiple
         replicas, documented on ES documentation.
         See: https://www.elastic.co/guide/en/elasticsearch/guide/current
         /_search_options.html#_preference for more information.
 
-        :param preference: A function that returns the preference value.
+        :param str preference: A preference value.
         """
-        return self.params(preference=preference)
-
+        if preference:
+            return self.params(preference=preference)
+        return self
```

Comment:
Just to understand, we want this in order to use chain operations (therefore need to return itself). Before the point was "preference is required, cuz otherwise there is no point on calling the function".

(Asking cuz I did not see chaining on invenio-records-resources)

Reply:
Doing the conditional and returning self is in part to keep a [similar interface to the previous](https://github.com/inveniosoftware/invenio-search/pull/207/files#diff-a63511cc73e3675c8dfe54c85f16ce135f92714bcca9f0c3153f4f25477c0ff4R161) `with_preference_param` and in other part because records-resources [was doing that check](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/records/service.py#L96), so might as well move it back inside the function and alleviate any caller from having to do that check.